Sources/Core/AWSClientRuntime/Auth/AuthSchemes/SigV4Util.swift

```diff
@@ -0,0 +1,46 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Foundation
+import ClientRuntime
+
+public class SigV4Util {
+    // List of services that use signed body heaader.
+    static let usesSignedBodyHeader = ["S3", "Glacier", "S3 Control"]
+    // Map of service::operation that use unsigned body for presign URL flow.
+    static let forceUnsignedBodyForPresigningURL = [
+        "S3": ["getObject", "putObject"]
+    ]
```

Comment:
Same as `AWSClientConfigDefaultsProvider.swift`, these things look like they should be customizations in codegen rather than in the Swift runtime package.

Reply:
These _can_ be moved to codegen if we generate the code that sets these flags based on runtime values of service name and operation name. But I figured having this in runtime, which is where the bulk of signing config customizations exist (e.g., under `SigV4AuthScheme::customizeSigningProperties`), would be better for code organization & following where and how signing properties are set.

During codegen the only signing related flags set are signing name & signing region and couple other flags that MUST be set during codegen such as what "flow" the generated operation stack is in (e.g., PRESIGN_URL vs. NORMAL).

In addition regardless of where it is, in terms of maintenance, it's the same either way in that the list of service names and operation names will need to be manually maintained, albeit with semi-permanence since services won't be changing how they authenticate requests all that often.

Reply:
I'm not opposed to keeping it there, but I'm wondering if there is something missing here. `customizeSigningProperties()` should only include the necessary signing property customizations needed for a given auth scheme.

The additional service and operation context (e.g. `usesSignedBodyHeader `, `forceUnsignedBodyForPresigningURL`, `setS3SpecificFlags`) seem like customizations scoped for specific services, but would be hardcoded in the generic `SigV4AuthScheme::customizeSigningProperties()`.

For the services `S3`, `Glacier`, `S3 Control`, I think it'd make more sense to:

1. Establish the generic `SigV4AuthScheme::customizeSigningProperties` method
2. Find some way to customize that method for these services specifically through codegen

Reply:
I see what you mean. Unless I'm mistaken, you're suggesting factoring out signing config logic that are not specific to an auth scheme but rather specific to the service to a separate location. 

Given a single generic auth scheme middleware is shared among services in the new Swift SDK auth structure instead of being code-generated for each service like in SRA (which was done to reduce generated code due to Xcode limitation), one approach I could take that mitigates this mixing of concerns is factoring out the flag-setting behaviors from `SigV4AuthScheme` and `SigV4AAuthScheme` entirely to `SigV4Util` instead of just using `Sigv4Util` as utility helper for getting conditions to set service-specific signing flags. 

In turn, the generic `customizeSigningProperties` method would instead have something like `return SigV4Util.customizeServiceSpecificSigningConfig()` at the end of the method body, instead of having hard-coded service specific config logic inside it.

Reply:
Had an off-line discussion, will refactor service specific config flag setting to either codegen or runtime that overrides default auth scheme implementations (e.g., SigV4AuthScheme, SigV4AAuthScheme).